docs/targets/solana.rst

```diff
@@ -1,23 +1,52 @@
-
-Solana has the following differences to Ethereum Solidity:
-
-- The address type is 32 bytes, not 20 bytes. This is what Solana calls an "account"
-- An address literal has to be specified using the ``address"36VtvSbE6jVGGQytYWSaDPG7uZphaxEjpJHUUpuUbq4D"`` syntax
-- There is no ``ecrecover()`` builtin function, but there is a ``signatureVerify()`` function which can check ed25519
-  signatures.
-- There is no concept of gas, so there is no gas functions
-- Solana balance is stored in a ``uint64``, so *address* ``.balance``, ``.transfer()`` and ``.send()``
-  all use ``uint64`` rather than ``uint256``.
-- Solana uses different accounts for the program code, and the contract data.
-- Programs are upgradable. There is no need to implement upgrades in Solidity.
+Solana Overview
+_______________
+
+The Solana blockchain has a unique account model that resembles a key-value database.
+As the underlying environment follows a programming model different than that of Ethereum, Solidity inner workings
+have been modified to work properly on Solana.
+
+A Solidity contract on Solana utilizes two accounts: a data account and a program account.
+The program account stores the contract's executable binary and owns the data account,
+which holds all the storage variables.
+
+Contract upgrades
++++++++++++++++++
+
+Provided that the data layout from a new contract is compatible with that of an old one, it is possible
+to update the binary in the program account and retain the same data, rendering
+contract upgrades implemented in Solidity unnecessary.
+
+Data types
+++++++++++
+
+- An account address consists of a 32-bits key, which is represented by the ``address`` type. This data model differs from
+  Ethereum 20-bits addresses.
+- Solana's virtual machine registers are 64-bit wide, so 64-bit integers ``uint64`` and ``int64`` are preferable over
+  ``uint256`` and `int256`. Likewise, the builtin functions for *address* ``.balance``, ``.transfer()`` and ``.send()``
+  use 64-bit integers.
+- An address literal has to be specified using the ``address"36VtvSbE6jVGGQytYWSaDPG7uZphaxEjpJHUUpuUbq4D"`` syntax.
+
+Runtime
++++++++
+
+- Each instruction on Solana has a fixed amount of compute units. This concept supersedes the idea of gas from
+  Ethereum, so gas related functions are not available on Solana.
+- There is no ``ecrecover()`` builtin function because Solana does not use the ECDSA algorithm, but there
+  is a ``signatureVerify()`` function which can check ed25519 signatures. As a consequence, it is not possible to
+  recover a signer from a signature.
+- The Solana target requires `Solana <https://www.solana.com/>`_ v1.8.1.
+- Function selectors are eight bytes wide and known as *discriminators*.
 - Solana provides different builtins, e.g. ``tx.program_id`` and ``tx.accounts``.
-- Function selectors are eight bytes and known as *discriminators*.
+- Try-catch statements do not work on Solana. If any invocation fails, the runtime will halt the contract.
```

Comment:
- invocation is not a term used by ethereum or solana and there it is ambiguous. 
- the execution is halted, not the contract (how is a contract halted)

Reply:
> * there it is ambiguous.

Which different meanings can you spot in the sentence?

Reply:
invocation could be be transaction, instruction, internal function call, external function call for example

Reply:
All those meanings are correct for the sentence, aren't they?
If a transaction, instruction, internal function call or external function call fail, the execution will halt.

Reply:
It's still very ambiguous. It's not clear at all what you mean.

Reply:
> * invocation is not a term used by ethereum or solana and there it is ambiguous.

CPI means "cross program invocation".

Reply:
> It's still very ambiguous. It's not clear at all what you mean.

It might be ambiguous, but you could pinpoint all the intended meanings, so it is not unclear.

Reply:
You're being pedantic. It is not clear to users.

Reply:
I have changed `invocation` (which is a term used by Solana - recall CPI) to `external call or contract creation`, which are the constructions allowed in a try-catch statement.

Reply:
The terminology which is used by Solana is CPI, not invocation.